web/javascript/transmission.js

```diff
@@ -1304,7 +1309,36 @@ Transmission.prototype = {
     reannounceTorrents: function (torrents) {
         this.remote.reannounceTorrents(this.getTorrentIds(torrents), this.refreshTorrents, this);
     },
-
+    showmagnetlinkTorrents: function(torrents) {
+        for (i = 0; t = torrents[i]; ++i) {
+            if (typeof t.getHashString() != "string" || t.getHashString().length != 40 || typeof t.getTrackers() != "object" || t.getTrackers().length < 1) {
+                break;
+            };
+            if (i == torrents.length-1) {
+                var data = [];
+                for (i = 0; t = torrents[i]; ++i) {
+                    data.push("magnet:?xt=urn:btih:" + t.getHashString() + "&dn=" + encodeURIComponent(t.getName()) + "&tr=" + t.getTrackers().map(function(e){return encodeURIComponent(e.announce);}).join('&tr='));
+                };
+                this.showMagnetLinks(torrents, data);
+                return;
+            };
+        };
+        this.remote.getMagnetLinks(this.getTorrentIds(torrents), function(data) {
+            this.showMagnetLinks(torrents, Array.from(data.torrents, x => x.magnetLink));
+        }, this);
+		
+    },
+    showMagnetLinks: function(torrents, data) {
+        $('#magnet-links-dialog > div').html('');
+        for (i = 0; t = torrents[i]; ++i) {
+            $('#magnet-links-dialog > div').append('<div class="title">' + t.getName() + '</div><div class="row"><input type="text" value="' + data[i] + '" readonly/></div>');
```

Comment:
looks to me like both `t.getName()` and `data[i]` *should* be html encoded here, but isn't, or am i missing something?

if there is no html encoding function available, perhaps something like

```js
function tohtml(text) {
    if (!tohtml.enc) {
        tohtml.enc = document.createElement("span");
    }
    tohtml.enc.textContent = text;
    return tohtml.enc.innerHTML.replace(/\"/g, "&quot;").replace(/\'/g, "&#039;");
}
```

(a quick google search suggests the internet cannot agree on how to html-encode a text string with javascript, this is my take on it.)

- another option is to create the entire thing with DOM functions